napari/layers/utils/_link_layers.py

```diff
@@ -0,0 +1,203 @@
+            )
+    else:
+        # if no attributes are specified, ALL valid attributes are linked.
+        attr_set = valid_attrs
+
+    # now, connect requested attributes between all requested layers.
+    links = []
+    for (lay1, lay2), attr in product(permutations(layers, 2), attr_set):
+
+        key = _link_key(lay1, lay2, attr)
+        # if the layers and attribute are already linked then ignore
+        if key in _LINKED_LAYERS:
+            continue
+
+        def _make_l2_setter(eq_op, l1=lay1, l2=lay2, attr=attr):
+            # where `eq_op` is an "equality checking function" that is suitable
+            # for the attribute object type.  (see ``get_equality_operator``)
+
+            def setter(event=None):
+                new_val = getattr(l1, attr)
+                # this line is the important part for avoiding recursion
+                if not eq_op(getattr(l2, attr), new_val):
+                    setattr(l2, attr, new_val)
+
+            setter.__doc__ = f"Set {attr!r} on {lay2} to that of {lay1}"
+            setter.__qualname__ = f"set_{attr}_on_layer_{id(lay2)}"
+            return setter
+
+        # get a suitable equality operator for this attribute type
+        eq_op = get_equality_operator(getattr(lay1, attr))
+        # acually make the connection
+        callback = _make_l2_setter(eq_op)
+        emitter_group = getattr(lay1.events, attr)
+        emitter_group.connect(callback)
+
+        # store the connection so that we don't make it again.
+        # and save an "unlink" function for the key.
+        _LINKED_LAYERS[key] = partial(emitter_group.disconnect, callback)
+        links.append(key)
+    return links
+
+
+def unlink_layers(layers: Iterable['Layer'], attributes: Iterable[str] = ()):
+    """Unlink previously linked ``attributes`` between all layers in ``layers``.
+
+    Parameters
+    ----------
+    layers : Iterable[napari.layers.Layer]
+        The list of layers to unlink.  All combinations of layers provided will
+        be unlinked.  If a single layer is provided, it will be unlinked from
+        all other layers.
+    attributes : Iterable[str], optional
+        The set of attributes to link.  If not provided, all connections
+        between the provided layers will be unlinked.
+    """
+    layer_ids = [id(layer) for layer in layers]
+    if not layer_ids:
+        raise ValueError("Must provide at least one layer to unlink")
+    elif len(layer_ids) == 1:
+        layer_id = layer_ids[0]
+        keys = (k for k in list(_LINKED_LAYERS) if layer_id in k[:2])
+    else:
+        layer_combos = {frozenset(i) for i in combinations(layer_ids, 2)}
+        keys = (k for k in list(_LINKED_LAYERS) if set(k[:2]) in layer_combos)
+    if attributes:
+        keys = (k for k in keys if k[2] in attributes)
+    _unlink_keys(keys)
+
+
+@contextmanager
+def linked_layers(layers: Iterable['Layer'], attributes: Iterable[str] = ()):
```

Comment:
hmm i wonder about this name - when I first saw it I thought of the `_LINKED_LAYERS`. @tlambert03 in zulip just suggested `layers_linked`